### PR TITLE
[+] Routes - Expose the record services and the permissions middleware

### PR DESCRIPTION
**Pull request by VincentMolinie**
## Pull Request checklist:

- [ ] Write an explicit title for the Pull Request
- [ ] Write changes made in the CHANGELOG.md
- [ ] Test manually the implemented changes
- [ ] Review my own code (indentation, syntax, style, simplicity, readability)
- [ ] Wonder if you can improve the existing code

### CODE REVIEW

File: src/serializers/resource.js [hunk 1]
```diff
@@ -12,7 +12,6 @@ function ResourceSerializer(
   model,
   records,
   integrator,
-  opts,
```
Comment (VincentMolinie) [inline src/serializers/resource.js]: This was unused

**Comment by VincentMolinie** [comment]
[(13) Smart Routes - Expose liana methods](https://trello.com/c/c7xhPwak/3869-13-smart-routes-expose-liana-methods)

File: src/index.js [hunk 2]
```diff
@@ -76,22 +79,30 @@ exports.ensureAuthenticated = (request, response, next) => {
...
     });
 }
 
 exports.init = (Implementation) => {
   const { opts } = Implementation;
+
+  configStore.Implementation = Implementation;
+  configStore.lianaOptions = opts;
+
   if (opts.onlyCrudModule === true) {
```
Comment (larcin) [inline src/index.js]: [Misc] What is the `onlyCrudModule` exactly?

File: src/index.js [hunk 3]
```diff
@@ -188,20 +197,37 @@ exports.init = (Implementation) => {
...
+        configStore.lianaOptions,
+      ).perform();
+      new ActionsRoutes(
+        app,
+        model,
+        configStore.Implementation,
+        configStore.integrator,
+        configStore.lianaOptions,
+      ).perform();
+
```
Comment (larcin) [inline src/index.js]: [Code Style] Unnecessary newline

File: src/middlewares/permissions.js [hunk 4]
```diff
@@ -19,24 +31,34 @@ function createCheckPermission(environmentSecret, collectionName) {
...
-  return {
-    checkPermission,
-    checkPermissionListAndSearch,
-  };
+  export() {
+    return this._checkPermission('export');
+  }
+
+  details() {
+    return this._checkPermission('show');
```
Comment (larcin) [inline src/middlewares/permissions.js]: [Misc] Why did you change from `show` to `details` ?

File: src/middlewares/permissions.js [hunk 5]
```diff
@@ -1,15 +1,27 @@
 const PermissionsChecker = require('../services/permissions-checker');
 const httpError = require('http-errors');
 const logger = require('../services/logger');
+const ConfigStore = require('../services/config-store');
 
 const getRenderingIdFromUser = user => user.renderingId;
 
-function createCheckPermission(environmentSecret, collectionName) {
-  function checkPermission(permissionName) {
+class PermissionMiddleWareCreator {
```
Comment (larcin) [inline src/middlewares/permissions.js]: [Wording] I kinda have an issue with the name `PermissionMiddleWareCreator` why wouldn't `PermissionsMiddleWare` be sufficient ?

Comment (larcin) [inline src/middlewares/permissions.js]: [Wording] middleware is a full word thus the camel case isn't used correctly here: `PermissionMiddlewareCreator `

File: src/services/exposed/record-updater.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+const ResourceDeserializer = require('../../deserializers/resource');
+
+class RecordUpdater extends AbstractRecordService {
+  update(record, recordId) {
+    return new this.Implementation.ResourceUpdater(this.model, { recordId }, record)
+      .perform();
+  }
+
+  deserialize(body) {
+    return new ResourceDeserializer(this.Implementation, this.model, body, false)
```
Comment (larcin) [inline src/services/exposed/record-updater.js]: [Misc] Why is there only the update that is done with `withRelationships` = false ?

File: src/index.js (hunk 2 above)
Comment (VincentMolinie) [reply src/index.js]: It was used by lumber-graphql to generate the crud routes but without forestadmin other routes. Should be removed in another PR

File: src/middlewares/permissions.js (hunk 4 above)
Comment (VincentMolinie) [reply src/middlewares/permissions.js]: To be more explicit. Show means nothing to me, details is better (My point of view, feel free to contest it)

File: src/middlewares/permissions.js (hunk 5 above)
Comment (VincentMolinie) [reply src/middlewares/permissions.js]: Because it's not a middleware it's something that creates middleware. When just reading the code you might want to juste use `PermissionMiddleware` in an `app.use(PermissionMiddleware)`. Which is wrong

File: src/services/exposed/record-updater.js (hunk 6 above)
Comment (VincentMolinie) [reply src/services/exposed/record-updater.js]: Old codes. I did not change anything on that. But to be short, when you create an entry you can create it with relationships (As you may require a record to have the relationship set to be able to create it). But when you update a record it updates only the record, not the relationships
